lark/parser_frontends.py

```diff
@@ -233,10 +228,19 @@ def _apply_callback(self, tree):
         return self.callbacks[tree.rule](tree.children)
 
 
-class MakeParsingFrontend(MakeParsingFrontend):
-    def __call__(self, lexer_conf, parser_conf, options):
-        assert isinstance(lexer_conf, LexerConf)
-        assert isinstance(parser_conf, ParserConf)
-        parser_conf.parser_type = self.parser_type
-        lexer_conf.lexer_type = self.lexer_type
-        return ParsingFrontend(lexer_conf, parser_conf, options)
+_parser_creators['earley'] = create_earley_parser
+_parser_creators['cyk'] = CYK_FrontEnd
+
+
+def _construct_parsing_frontend(
```

Comment:
`MakeParsingFrontend.__call__()` was using the instance variables. However the only use of a class was in a single function, which immediately used it as a `Callable`. So I feel like it is cleaner to make all of the values arguments to a single function.